File: app/src/main/java/com/github/aakumykov/sync_dir_to_cloud/view/interfaces/iSyncTaskStarter.java
package com.github.aakumykov.sync_dir_to_cloud.view.interfaces;

import com.github.aakumykov.sync_dir_to_cloud.view.domain.entities.SyncTask;

public interface iSyncTaskStarter {

    void startSyncTask(final SyncTask syncTask, Callbacks callbacks);

    interface Callbacks {
        void onSyncTaskStarted();
        void onSyncTaskStartingError(Exception e);
    }
}

File: app/src/main/java/com/github/aakumykov/sync_dir_to_cloud/interfaces/iSyncTaskStarter.kt
package com.github.aakumykov.sync_dir_to_cloud.interfaces


import com.github.aakumykov.sync_dir_to_cloud.domain.entities.SyncTask

interface iSyncTaskStarter {

    fun startSyncTask(syncTask: SyncTask, callbacks: Callbacks)

    interface Callbacks {
        fun onSyncTaskStarted()
        fun onSyncTaskStartingError(e: Exception)
    }
}
